### PR TITLE
feat: batch operations and bulk operations

### PR DESCRIPTION
**Pull request by jo-jordan**
Add methods for supporting bulk operations and batch operations.

- batchCreate
- batchUpsert
- batchDelete
- bulkCreate
- bulkUpsert
- bulkDelete

Note: batch is transactional and bulk is not transactional.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,7 +139,39 @@ db.upsert("Collection1", user1, "Users");
...
+db.batchUpsert("Collection1", List.of(user1), "Users");
+
+// Delete
+db.batchDelete("Collection1", List.of(user1), "Users");
+// or
+db.batchDelete("Collection1", List.of(user1.id), "Users");
+
+```
 
+### Bulk Operation
```
Comment (thunderz99) [inline README.md]: Please write the difference between batch operation and bulk operation in the README🙏.  And add the official link to be more helpful to whom wants to know the details.

```
e.g.
https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/transactional-batch?tabs=java
```

File: src/main/java/io/github/thunderz99/cosmos/dto/CosmosBulkResult.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    /**
+     * The result of retries operations that exceed the max retry times
+     */
+    public List<?> retryList = new ArrayList<>();
+
+    /**
+     * The result of fetal operations. e.g. 409 Conflict
+     */
+    public List<CosmosException> fetalList = new ArrayList<>();
```
Comment (thunderz99) [inline src/main/java/io/github/thunderz99/cosmos/dto/CosmosBulkResult.java]: Spell miss: `fetal`. Should be `fatal` or `failed`, etc

Comment (jo-jordan) [reply src/main/java/io/github/thunderz99/cosmos/dto/CosmosBulkResult.java]: Oh oops. I've used `fetal` as 'fatal' for a long time. 🥶

File: src/main/java/io/github/thunderz99/cosmos/CosmosDatabase.java [hunk 3]
```diff
@@ -89,6 +98,294 @@ public CosmosDocument create(String coll, Object data, String partition) throws
...
+
+        doBatchWithRetry(container, batch);
+
+        return ids.stream().map(it -> {
+            var doc = new Document(JsonUtil.toJson(Map.of("id", it)));
+            return new CosmosDocument(doc.toObject(JSONObject.class));
+        }).collect(Collectors.toList());
+    }
+
+    private static String getId(Object object) {
```
Comment (thunderz99) [inline src/main/java/io/github/thunderz99/cosmos/CosmosDatabase.java]: Please change private to package and add unit test for this method.

File: src/main/java/io/github/thunderz99/cosmos/CosmosDatabase.java [hunk 4]
```diff
@@ -89,6 +98,294 @@ public CosmosDocument create(String coll, Object data, String partition) throws
...
+        if (object instanceof String) {
+            id = (String) object;
+        } else {
+            var map = JsonUtil.toMap(object);
+            id = map.get("id").toString();
+        }
+        return id;
+    }
+
+    private static void doCheckBeforeBatch(String coll, List<?> data, String partition) {
```
Comment (thunderz99) [inline src/main/java/io/github/thunderz99/cosmos/CosmosDatabase.java]: Please change private to package and add unit test for this method (include boundary checks).

File: src/main/java/io/github/thunderz99/cosmos/CosmosDatabase.java [hunk 5]
```diff
@@ -89,6 +98,294 @@ public CosmosDocument create(String coll, Object data, String partition) throws
...
+        Checker.checkNotBlank(coll, "coll");
+        Checker.checkNotBlank(partition, "partition");
+        Checker.checkNotEmpty(data, "create data " + coll + " " + partition);
+
+        checkBatchMaxOperations(data);
+        checkValidId(data);
+    }
+
+    private static void doCheckBeforeBulk(String coll, List<?> data, String partition) {
```
Comment (thunderz99) [inline src/main/java/io/github/thunderz99/cosmos/CosmosDatabase.java]: Please change private to package and add unit test for this method (include boundary checks).

File: src/main/java/io/github/thunderz99/cosmos/CosmosDatabase.java [hunk 6]
```diff
@@ -89,6 +98,294 @@ public CosmosDocument create(String coll, Object data, String partition) throws
...
+
+    static void checkBatchMaxOperations(List<?> data) {
+        // There's a current limit of 100 operations per TransactionalBatch to ensure the performance is as expected and within SLAs:
+        // https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/transactional-batch?tabs=dotnet#limitations
+        if (data.size() > MAX_BATCH_NUMBER_OF_OPERATION) {
+            throw new IllegalArgumentException("The number of data operations should not exceed 100.");
+        }
+    }
+
+    static void checkValidId(List<?> data) {
```
Comment (thunderz99) [inline src/main/java/io/github/thunderz99/cosmos/CosmosDatabase.java:378]: Please add unit test for this method.

Comment (jo-jordan) [reply src/main/java/io/github/thunderz99/cosmos/CosmosDatabase.java:378]: All four unit tests have been added.
